File: ECnotes/Sem1/Labs/LivingExamples/VB/Ch10Navigation/MasterPage.master.vb

Partial Class MasterPage
    Inherits System.Web.UI.MasterPage

    Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        Dim iDaysUntil As Integer = DaysUntilHalloween()
        If iDaysUntil = 0 Then
            lblMessage.Text = "Happy Halloween!"
        ElseIf iDaysUntil = 1 Then
            lblMessage.Text = "Tomorrow is Halloween!"
        Else
            lblMessage.Text = "There are only " & iDaysUntil _
                & " days left until Halloween!"
        End If
    End Sub

    Private Function DaysUntilHalloween() As Integer
        Dim dtmHalloween As Date = New DateTime(DateTime.Today.Year, 10, 31)
        If DateTime.Today > dtmHalloween Then
            dtmHalloween = dtmHalloween.AddYears(1)
        End If
        Dim tsTimeUntil As TimeSpan = dtmHalloween - DateTime.Today
        Return tsTimeUntil.Days
    End Function

End Class

File: ECnotes/Sem1/Labs/LivingExamples/CS/Ch10Navigation/MasterPage.master.cs
using System;
using System.Data;
using System.Configuration;
using System.Collections;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;

public partial class MasterPage : System.Web.UI.MasterPage
{
    protected void Page_Load(object sender, EventArgs e)
    {
        int daysUntil = DaysUntilHalloween();

        if (daysUntil == 0)
        {
            lblMessage.Text = "Happy Halloween!";
        }
        else if (daysUntil == 1)
        {
            lblMessage.Text = "Tomorrow is Halloween!";
        }
        else
        {
            lblMessage.Text = "There are only " + daysUntil
                + " days left until Halloween!";
        }
    }

    private int DaysUntilHalloween()
    {
        DateTime halloween = new DateTime(DateTime.Today.Year, 10, 31);

        if (DateTime.Today > halloween)
        {
            halloween = halloween.AddYears(1);
        }

        TimeSpan ts = halloween - DateTime.Today;

        return ts.Days;
    }
}
